connect/src/main/scala/org/apache/spark/sql/connect/package.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
+
+import org.apache.spark.connect.proto
+
```

Comment:
Should probably document this like https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala#L36-L64. BTW since we're here, can we match the DSL object/class structure w/ Catalyst one?

Reply:
I changed to match what catalyst's DSL is doing. I also added  class comment.

However when I try to document the usage guide, I found the import that works in test code which does not work for spark-shell

```

scala> import org.apache.spark.sql.connect.plans.DslLogicalPlan
<console>:23: error: object connect is not a member of package org.apache.spark.sql
       import org.apache.spark.sql.connect.plans.DslLogicalPlan
```

 I am thinking this is a part of cleanup work that we need to do. Can I create a follow up JIRA to fix this then go back to document the DSL usage next time?

Reply:
I double-checked in case I did't include right DSL classes in the shell but just import a different connect class with same error:
```
scala> import org.apache.spark.sql.connect.service.SparkConnectService
<console>:25: error: object connect is not a member of package org.apache.spark.sql
       import org.apache.spark.sql.connect.service.SparkConnectService
```

Reply:
Ah I found that we need to add connect to the deps of the REPL module. 

I made the change and also then update the documentation.